src/app/chat/views/chat_members.nim

```diff
@@ -0,0 +1,57 @@
+import NimQml, Tables
+import ../../../status/chat/[chat, message]
+
+type
+  ChatMemberRoles {.pure.} = enum
+    UserName = UserRole + 1,
+    PubKey = UserRole + 2,
+    IsAdmin = UserRole + 3,
+    Joined = UserRole + 4
+    Identicon = UserRole + 5
+
+QtObject:
+  type
+    ChatMembersView* = ref object of QAbstractListModel
+      members*: seq[ChatMember]
+
+  proc setup(self: ChatMembersView) = self.QAbstractListModel.setup
+
+  proc delete(self: ChatMembersView) = self.QAbstractListModel.delete
+
+  proc newChatMembersView*(): ChatMembersView =
+    new(result, delete)
+    result.members = @[]
+    result.setup()
+
+  proc setMembers*(self: ChatMembersView, members: seq[ChatMember]) =
+    self.beginResetModel()
+    self.members = members
+    self.endResetModel()
+
+  proc len*(self: ChatMembersView): int {.slot.} = self.members.len
```

Comment:
This function is not super useful, you can just use `rowCount`. You can add the slot macro if you need it in the QML